Adding XIAO ESP32C6 with BLE support

alexbegoon opened the pull request:
Related to https://github.com/meshtastic/firmware/pull/4692
The main purpose of this PR is to enhance the ESP32-C6 target by integrating BLE functionality

alexbegoon commented on protobufs:
My bad! :cold_sweat:  I didn't mean to do that. Resetting now.

caveman99 replied:
Please don't take that as being offensive :-) It's just these files (and the submodule especially) are notorious for generating merge conflicts later on and they're auto-generated anyway.

alexbegoon commented on bin/version_script.py:
For some reason, the call to `platformio-custom.py` and the following part:
```python
# General options that are passed to the C and C++ compilers
projenv.Append(
    CCFLAGS=[
        "-DAPP_VERSION=" + verObj["long"],
        "-DAPP_VERSION_SHORT=" + verObj["short"],
        "-DAPP_ENV=" + env.get("PIOENV"),
    ]
)
```
has no effect. 

I tried adding the following lines to the end of `platformio-custom.py`:
```python
env.Append(
   BUILD_FLAGS=[
      "-DAPP_VERSION=" + verObj["long"],
      "-DAPP_VERSION_SHORT=" + verObj["short"],
      "-DAPP_ENV=" + env.get("PIOENV"),
   ]
)
```
but it didn't work either. 

What eventually worked was [adding this line](https://github.com/meshtastic/firmware/pull/5421/files#diff-cb2ce30e87a30bf39576ce855d63f653548f69a4743af973d9eb71b34d3d3222R12) with the `pre` instruction:
```yaml
extra_scripts = pre:bin/version_script.py
```
It seems that ESP-IDF adds its own `CMakeLists.txt` to the root of the project during compilation, which might cause the root `platformio.ini` to become disconnected.

Maybe another pair of eyes can help here. :slightly_smiling_face:

caveman99 replied:
No problem, i just hate duplicate code. i am not python savvy but maybe we can 'include' the new file in the existing one, replacing that part?

alexbegoon replied:
Agree. I'll refactor this. 

Just for the reference I'll leave it here: https://github.com/platformio/platformio-core/issues/4329#issuecomment-1245553192

caveman99 commented:
I am testing a few corner cases with this. is it ok to push minor corrections to your branch?

alexbegoon commented:
> I am testing a few corner cases with this. Especially with the other C6 board (TLora C6) we have. Is it ok to push minor corrections to your branch?

Sure, I am fine with that!

alexbegoon commented:
@caveman99 just FYI, I'm fixing the failed builds for other targets... :shrug:

alexbegoon commented:
Hi @caveman99 ,

I think we're good to go now. Could you take a look please?

caveman99 commented on CMakeLists.txt:
Can we add a comment here that this is only for BLE support on ESP-IDF5? Presense of this file may lead people to other conclusions :-)

alexbegoon replied:
Added :slightly_smiling_face:

caveman99 commented:
Status: it compiles again but i don't see any BLE advertising emitted from my TLora C6. I have Seeed Xiao C6 incoming in about 10 days to counter-check.

Another thing i noticed when working with VSCode on Windows, due to the different file name handling compared to Linux the wrong RTC.h and CCM.h headers are picked up, presumably in ESP-IDF directories. I could remedy this for the RTC.h by prefixing the dir, but the crypto header is not so easy. I hope i can come up with a fix for this.

caveman99 commented:
Bluetooth is indeed enabled per info dump. I was scanning with the Android "nRF52 connect" app for any bluetooth signal from the device and wasn't lucky. i'll compare the sdkconfig later when i am back home.

```
"bluetooth": {
    "enabled": true,
    "mode": "FIXED_PIN",
    "fixedPin": 123456
  },
```

caveman99 commented:
I have a xiao C6 here now and can confirm it's advertising after boot. shame it doesn't find a radio but that's not important for this use case :-) Now to find out why the Tlora-C6 doesn't advertise.

By the way, there's excessive logspam from the ESP-IDF libs, every time a GPIO toggles it locks. Is there a way to disable that during build?

alexbegoon commented:
Hi @caveman99,

Thank you for confirming on your side. We are one step closer.

First of all, take a look at [this new flag](https://github.com/meshtastic/firmware/pull/5421/files#diff-a2be81c494cea5929f8ca266211533599aaaa0f146c9522a549ceb81407655e0R23). If it is applied, then a 2.4GHz antenna must be attached to the module to avoid damaging the RF output. :warning: I added this to improve BLE coverage.

> Shame it doesn't find a radio, but that's not important for this use case :-)

I’m not sure which radio you're talking about, but I can help you with the wiring if needed. I currently use a cheap AliExpress module with an SX1276, but I plan to switch to a Wio SX1262 soon (the PCB is already in development).

> By the way, there's excessive log spam from the ESP-IDF libs; every time a GPIO toggles, it logs. Is there a way to disable that during the build?

Yes, you're right. I fixed it [here](https://github.com/meshtastic/firmware/pull/5421/files#diff-4be82613e414f7d1ce2ed71901a542a1ffc9407da70f96682c30c6084d78ebfbR112). Now, the log level for GPIO is set to WARN.

> Now to find out why the Tlora-C6 doesn't advertise.

Yes, this is interesting one. I'll check if this module have some sort of balun/rf-switch that has to be managed by a gpio pin.

alexbegoon commented:
Hi @caveman99 ,

I checked online, and the best I could find is this [schematic](https://github.com/Xinyuan-LilyGO/LilyGo-LoRa-Series/blob/master/schematic/T3-C6_V1.0.pdf). Unfortunately, it seems the RF frontend is grounded there. I don’t have this module to verify, but I’m quite sure the only antenna it has is a sub-ghz antenna. :man_shrugging: 

I don’t believe this module can support a 2.4 ghz connection.

<img src="https://github.com/user-attachments/assets/4461bef9-b1cc-405e-a1a5-5b4f2ae5ff55"  width="500px" />

weekroom commented:
@alexbegoon Hello.
I modified part of the code on the basis of your code, so that the Bluetooth and communication of the device can be found normally in the window system. Currently, I can communicate normally when using an Android phone during the test, but when using an Apple phone, the Bluetooth connection with the device cannot be made, so it is suddenly disconnected. May I ask if you also have such a problem? Here is my Android screen interface (f5ec is my esp32c6 device, e958 is my esp32s3 device)

![0f1b00dc3434d07d0f8b4b6a4e4377a](https://github.com/user-attachments/assets/40087de4-fb6c-4aa0-9bbe-0d721c8ce036)

alexbegoon commented:
Hello @weekroom ,
That's great that you got it working!
On my end, it's operating just fine with the Android client. I can't confirm the situation with an iPhone. IIRC, from some Discord discussions, this is a common issue with iPhones due to their aggressive/greedy handling of long-lived connections.

weekroom commented:
@alexbegoon hello，
      I have modified the code in the NimbleBluetooth.cpp file about interaction with Bluetooth. At present, I can connect and communicate with an Android phone and an Apple phone. I added the device's Bluetooth name to the broadcast packet in the startAdvertising function, which made it easier for the iPhone interface to display the device's Bluetooth name. Without this operation, the app of the iPhone would be unknown. I hope you find my code useful.
[NimbleBluetooth.txt](https://github.com/user-attachments/files/18703407/NimbleBluetooth.txt)

![c75eaf3841efd7a67c7a9454c97041e](https://github.com/user-attachments/assets/d5a29661-9bda-401a-a599-4a2b5bb16b39)
![29acd64ce105f51356b0f1ce8eab8c3](https://github.com/user-attachments/assets/1046083b-69a6-4aa3-a0a5-62e13121165e)

alexbegoon commented:
Hi @weekroom ,

I checked the file, but it seems a bit outdated compared to the current master. Could you please apply your BLE-related changes to this branch?

weekroom commented:
ble part and the main branch of the code is different, I am afraid I changed the code only support c6, I did not use other devices to test, so I used a shared file instead of merging in your branch, later I re-changed the code, and added macro definition to distinguish

alexbegoon committed:
Fix conflicts

alexbegoon commented:
On 2.6.0 the firmware is building but not sending data over BLE. Converting it back to a draft..

alexbegoon committed:
Merge remote-tracking branch 'upstream/master' into xiao_esp32c6_with_ble

 Conflicts:
	bin/platformio-custom.py
	src/main.cpp

alexbegoon committed:
bump deps version

JahazielLem commented:
Hi @alexbegoon
I'm trying to test this PR for my custom board based on the ESP32C6, but when I try to compile failed with the following error with with CTR lib:
src/mesh/CryptoEngine.h:81:5: error: 'CTRCommon' does not name a type; did you mean 'AESCommon'?
   81 |     CTRCommon *ctr = NULL;
      |     ^~~~~~~~~
      |     AESCommon
*** [.pio/build/seeed-xiao-esp32c6/src/PowerFSM.cpp.o] Error 1
*** [.pio/build/seeed-xiao-esp32c6/src/RedirectablePrint.cpp.o] Error 1

Have you this problem before?
You PR are the unique that does not compile me, the master branch works OK.
Regards

pierreyves258 commented:
> Status: it compiles again but i don't see any BLE advertising emitted from my TLora C6. I have Seeed Xiao C6 incoming in about 10 days to counter-check.
> 
> Another thing i noticed when working with VSCode on Windows, due to the different file name handling compared to Linux the wrong RTC.h and CTR.h headers are picked up, presumably in ESP-IDF directories. I did remedy this for the RTC.h by prefixing the dir, but the crypto engine header is not so easy. I hope i can come up with a fix for this.

Same here, on e esp32-c6-devkitc-1 I don't see BLE advertising and I got this message in logs
`ESP_ERROR_CHECK_WITHOUT_ABORT failed: esp_err_t 0x105 (ESP_ERR_NOT_FOUND) at 0x4203490c
file: "src/platform/esp32/BleOta.cpp" line 16
func: static const esp_partition_t* BleOta::findEspOtaAppPartition()
expression: esp_ota_get_partition_description(part, &app_desc)
ESP_ERROR_CHECK_WITHOUT_ABORT failed: esp_err_t 0x102 (ESP_ERR_INVALID_ARG) at 0x42034992
file: "src/platform/esp32/BleOta.cpp" line 30
func: static String BleOta::getOtaAppVersion()
expression: esp_ota_get_partition_description(part, &app_desc)`

alexbegoon committed:
Merge remote-tracking branch 'upstream/master' into xiao_esp32c6_with_ble

alexbegoon commented:
Hi @JahazielLem ,

I had just updated to the latest master and tried to build it at  https://github.com/meshtastic/firmware/pull/5421/commits/5336a1d747f1ee3cb6e6b14a1777948493a8d5a8. It worked fine on my end. I have never seen this error before.

<img src="https://github.com/user-attachments/assets/14bfc1ce-42d7-4a4c-99d7-d0f0c1ca2a04" width="300px"/>

alexbegoon committed:
Merge remote-tracking branch 'upstream/master' into xiao_esp32c6_with_ble

caveman99 committed:
Merge branch 'master' into xiao_esp32c6_with_ble

thebentern committed:
Merge branch 'master' into xiao_esp32c6_with_ble

caveman99 committed:
Merge branch 'master' into xiao_esp32c6_with_ble

caveman99 commented:
@alexbegoon can you have a look at https://github.com/meshtastic/firmware/pull/6510 ? this will just update the nimBLE stack to 2.x - it seems to have problems with actually showing the pairing pin, the rest works. If that one is in place, we don't have to macro-guard between the different ESP variants any more and can just add the ESP-IDF stuff needed for C6.

alexbegoon commented:
@caveman99 I'll check tomorrow. I'm quite positive because I recently updated this branch with https://github.com/h2zero/esp-nimble-cpp/releases/tag/2.0.2, and it works well with the 2nd version.

alexbegoon commented:
Hi @caveman99 I think this change will fix the PIN code issue: https://github.com/meshtastic/firmware/pull/6574/files
Could you please check with an ESP device on your end?

caveman99 committed:
Merge branch 'master' into xiao_esp32c6_with_ble

caveman99 committed:
Merge branch 'master' into xiao_esp32c6_with_ble

caveman99 commented on src/nimble/NimbleBluetooth.cpp:
```diff
@@ -122,7 +122,7 @@ class NimbleBluetoothServerCallback : public NimBLEServerCallbacks
 
-        return passkey;
+        NimBLEDevice::injectPassKey(connInfo, passkey);
     }
```
i think the return is right here. inject passkey is for user proovided input through a display or keyboard.

alexbegoon replied:
Correct. Fixed that part.

vidplace7 commented on variants/tlora_c6/platformio.ini:
```diff
@@ -1,10 +1,41 @@
 [env:tlora-c6]
 extends = esp32c6_base
+platform = https://github.com/pioarduino/platform-espressif32.git#f5d602fa7b4b778604306a149fe474723e79be2d
+framework = arduino, espidf
```
```suggestion
```
You should move these changes into the `esp32c6_base` in `arch/esp32/esp32c6.ini` :+1:

alexbegoon replied:
Done.

vidplace7 commented on variants/seeed_xiao_esp32c6/platformio.ini:
```diff
@@ -0,0 +1,40 @@
+  managed_components/espressif__esp_rainmaker/server_certs/rmaker_claim_service_server.crt
+  managed_components/espressif__esp_rainmaker/server_certs/rmaker_ota_server.crt
+
+lib_deps =
+  ${esp32c6_base.lib_deps}
+  https://github.com/h2zero/esp-nimble-cpp.git#2.2.1
+
```
```suggestion
```
Can this be moved to `arch/esp32/esp32c6.ini`?

alexbegoon replied:
Yep, moved it.

vidplace7 reviewed (changes requested):
I've just got platformio nitpicks for ya :sweat_smile:
Please try to move as much as you can from `variants/**/platformio.ini` to `arch/esp32/esp32c6.ini` to make this a bit more maintainable. :heart:

alexbegoon commented:
@caveman99 @vidplace7 I got your comments, I'll address them asap

alexbegoon committed:
Merge remote-tracking branch 'upstream/master' into xiao_esp32c6_with_ble

alexbegoon committed:
Improve pio config.
Fix Nimble pin code issue.
Address some PR comments.

alexbegoon committed:
Fix lint

alexbegoon committed:
Fix lint

alexbegoon commented on sdkconfig.defaults:
Added some comments.

caveman99 commented on arch/esp32/esp32c6.ini:
```diff
@@ -32,6 +30,7 @@ lib_deps =
   https://github.com/meshtastic/ESP32_Codec2/archive/633326c78ac251c059ab3a8c430fcdf25b41672f.zip
   # renovate: datasource=custom.pio depName=rweather/Crypto packageName=rweather/library/Crypto
   rweather/Crypto@^0.4.0
+  https://github.com/h2zero/esp-nimble-cpp.git#2.2.1
```
Hate to be a party pooper, but these GIT pulls are very expensive during build time so we replaced them with on-the-fly ZIP files. Please consider using https://github.com/h2zero/esp-nimble-cpp/archive/2.2.1.zip instead

vidplace7 replied:
```suggestion
  # renovate: datasource=github-tags depName=esp-nimble-cpp packageName=h2zero/esp-nimble-cpp
  https://github.com/h2zero/esp-nimble-cpp/archive/2.2.1.zip
```
Here's a complete version of what @caveman99 described :heart: with renovate

alexbegoon replied:
Yep, improved that part. Thank you!

alexbegoon committed:
Fix a dependency source

alexbegoon committed:
Merge remote-tracking branch 'upstream/master' into xiao_esp32c6_with_ble

Ope8888 commented:
Hello, can someone provide me with a pre-compiled firmware for ESP32C6?
Thanks.

alexbegoon committed:
Merge remote-tracking branch 'upstream/master' into xiao_esp32c6_with_ble

alexbegoon committed:
Add support for CONFIG_NIMBLE_CPP_IDF in NimbleBluetooth callbacks
Add support 1.x.x version of NimBLE-Arduino

alexbegoon commented:
@caveman99 @thebentern I [added support](https://github.com/meshtastic/firmware/pull/5421/commits/685c8c1df7f72b6238b42d4f9cceda22f64d0801) for new and older version of Arduino-NimBLE. This is related to this [commit](https://github.com/meshtastic/firmware/commit/48dc0e014c451fed8b3979d8548a8f328a613301)

Built and BLE work as expected at commit [685c8c1df](https://github.com/meshtastic/firmware/pull/5421/commits/685c8c1df7f72b6238b42d4f9cceda22f64d0801):
<img  src="https://github.com/user-attachments/assets/e22ecee0-171e-4176-b696-cebb829c5840" width="300px" />


@Ope8888 here is the pre-compiled version for XIAO ESP32C6 at commit [685c8c1df](https://github.com/meshtastic/firmware/pull/5421/commits/685c8c1df7f72b6238b42d4f9cceda22f64d0801)  ( :warning:  Don't forget to connect external antenna for 2.4ghz):
[firmware.zip](https://github.com/user-attachments/files/19964597/firmware.zip)

Ope8888 commented:
> @caveman99 @thebenternJ'ai [ajouté la prise en charge](https://github.com/meshtastic/firmware/pull/5421/commits/685c8c1df7f72b6238b42d4f9cceda22f64d0801) des nouvelles et anciennes versions d'Arduino-NimBLE. Ceci est lié à ce [commit.](https://github.com/meshtastic/firmware/commit/48dc0e014c451fed8b3979d8548a8f328a613301)
> 
> Construit et BLE fonctionne comme prévu lors du commit [685c8c1df](https://github.com/meshtastic/firmware/pull/5421/commits/685c8c1df7f72b6238b42d4f9cceda22f64d0801) : <img alt="" width="300px" src="https://private-user-images.githubusercontent.com/4046572/438887071-e22ecee0-171e-4176-b696-cebb829c5840.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU5ODM1NzYsIm5iZiI6MTc0NTk4MzI3NiwicGF0aCI6Ii80MDQ2NTcyLzQzODg4NzA3MS1lMjJlY2VlMC0xNzFlLTQxNzYtYjY5Ni1jZWJiODI5YzU4NDAuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDQzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA0MzBUMDMyMTE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2ZhYTE3ODA4YTQ2MGY5NzhkMTUxM2RjNmM4MDY5OGQ4MTZkNDVkZTBiNTk0OWRiYTY0NzYyY2JiZThmMDFjNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ONMVmbWBOwZ2TN6i3OjBWd0bTWcQVujTd7PS3hqE9pM">
> 
> @Ope8888voici la version précompilée pour XIAO ESP32C6 au commit [685c8c1df](https://github.com/meshtastic/firmware/pull/5421/commits/685c8c1df7f72b6238b42d4f9cceda22f64d0801) (⚠️ N'oubliez pas de connecter une antenne externe pour 2,4 GHz) : [firmware.zip](https://github.com/user-attachments/files/19964597/firmware.zip)

What flasher do you use?
I can't flash on my Xiao ESP32C6.
Thanks.

alexbegoon committed:
Reroute to Meshimi pinout
Add MAX17261 sensor

alexbegoon committed:
Add MAX17261 sensor

alexbegoon committed:
Merge remote-tracking branch 'upstream/master' into xiao_esp32c6_with_ble

Conflicts:
	arch/esp32/esp32c6.ini
	variants/tlora_c6/platformio.ini

renzenicolai commented:
@alexbegoon Thank you for putting in this effort, I'd love to see this PR be merged as that would open the door for me to add support for some hardware I have been working on which also uses the ESP32-C6 to run Meshtastic.

Ope8888 commented:
> > > @caveman99@thebenternJ'ai [ajouté la prise en charge](https://github.com/meshtastic/firmware/pull/5421/commits/685c8c1df7f72b6238b42d4f9cceda22f64d0801) des nouvelles et anciennes versions d'Arduino-NimBLE. Ceci est lié à ce [commit.](https://github.com/meshtastic/firmware/commit/48dc0e014c451fed8b3979d8548a8f328a613301) 
> > > Construit et BLE fonctionne comme prévu lors du commit [685c8c1df](https://github.com/meshtastic/firmware/pull/5421/commits/685c8c1df7f72b6238b42d4f9cceda22f64d0801) : 
> > > @Ope8888voici la version précompilée pour XIAO ESP32C6 au commit [685c8c1df](https://github.com/meshtastic/firmware/pull/5421/commits/685c8c1df7f72b6238b42d4f9cceda22f64d0801) (<img alt="" width="300px" src="https://private-user-images.githubusercontent.com/4046572/438887071-e22ecee0-171e-4176-b696-cebb829c5840.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU5ODM1NzYsIm5iZiI6MTc0NTk4MzI3NiwicGF0aCI6Ii80MDQ2NTcyLzQzODg4NzA3MS1lMjJlY2VlMC0xNzFlLTQxNzYtYjY5Ni1jZWJiODI5YzU4NDAuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDQzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA0MzBUMDMyMTE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2ZhYTE3ODA4YTQ2MGY5NzhkMTUxM2RjNmM4MDY5OGQ4MTZkNDVkZTBiNTk0OWRiYTY0NzYyY2JiZThmMDFjNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ONMVmbWBOwZ2TN6i3OjBWd0bTWcQVujTd7PS3hqE9pM">[](https://github.com/meshtastic/firmware/pull/5421/commits/685c8c1df7f72b6238b42d4f9cceda22f64d0801)⚠️N'oubliez pas de connecter une antenne externe pour 2,4 GHz) : [firmware.zip](https://github.com/user-attachments/files/19964597/firmware.zip)
> > 
> > 
> > J'ai réussi avec une de mes puces ;) mais la deuxième refuse, elle se connecte et se déconnecte en boucle. On peut la réinitialiser ? Merci.
> 
> Oui, vous pouvez réinitialiser manuellement. Consultez ce manuel : https://wiki.seeedstudio.com/xiao_esp32c6_getting_started/#reset

Hello, how do you flash? I succeeded in one of the three. I don't understand what my problem is.
The other two connect and disconnect in a loop, and I can't do anything with them, not even the reset.

Ope8888 commented:
> > > > @caveman99@thebenternJ'ai [ajouté la prise en charge](https://github.com/meshtastic/firmware/pull/5421/commits/685c8c1df7f72b6238b42d4f9cceda22f64d0801) des nouvelles et anciennes versions d'Arduino-NimBLE. Ceci est lié à ce [commit.](https://github.com/meshtastic/firmware/commit/48dc0e014c451fed8b3979d8548a8f328a613301)
> > > > Construit et BLE fonctionne comme prévu lors du commit [685c8c1df](https://github.com/meshtastic/firmware/pull/5421/commits/685c8c1df7f72b6238b42d4f9cceda22f64d0801) :
> > > > @Ope8888voici la version précompilée pour XIAO ESP32C6 au commit [685c8c1df](https://github.com/meshtastic/firmware/pull/5421/commits/685c8c1df7f72b6238b42d4f9cceda22f64d0801) (<img alt="" width="300px" src="https://private-user-images.githubusercontent.com/4046572/438887071-e22ecee0-171e-4176-b696-cebb829c5840.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU5ODM1NzYsIm5iZiI6MTc0NTk4MzI3NiwicGF0aCI6Ii80MDQ2NTcyLzQzODg4NzA3MS1lMjJlY2VlMC0xNzFlLTQxNzYtYjY5Ni1jZWJiODI5YzU4NDAuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDQzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA0MzBUMDMyMTE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2ZhYTE3ODA4YTQ2MGY5NzhkMTUxM2RjNmM4MDY5OGQ4MTZkNDVkZTBiNTk0OWRiYTY0NzYyY2JiZThmMDFjNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ONMVmbWBOwZ2TN6i3OjBWd0bTWcQVujTd7PS3hqE9pM">⚠️N'oubliez pas de connecter une antenne externe pour 2,4 GHz) : [firmware.zip](https://github.com/user-attachments/files/19964597/firmware.zip)
> > > 
> > > 
> > > J'ai réussi avec une de mes puces ;) mais la deuxième refuse, elle se connecte et se déconnecte en boucle. On peut la réinitialiser ? Merci.
> > 
> > 
> > Oui, vous pouvez réinitialiser manuellement. Consultez ce manuel : https://wiki.seeedstudio.com/xiao_esp32c6_getting_started/#reset
> 
> Hello, how do you flash? I succeeded in one of the three. I don't understand what my problem is. The other two connect and disconnect in a loop, and I can't do anything with them, not even the reset.

Debug: Sync err Error: Invalid head of packet (0x69): Possible serial noise or corruption.
Debug: Error at sync Error: Invalid head of packet (0x69): Possible serial noise or corruption.
Debug: Sync connect attempt 2
Debug: Sync
Debug: Sync err Error: Invalid head of packet (0x61): Possible serial noise or corruption.
Debug: Error at sync Error: Invalid head of packet (0x61): Possible serial noise or corruption.
Debug: Sync connect attempt 3
Debug: Sync
Debug: Sync err Error: Invalid head of packet (0x64): Possible serial noise or corruption.
Debug: Error at sync Error: Invalid head of packet (0x64): Possible serial noise or corruption.
Debug: Sync connect attempt 4

caveman99 committed:
Merge branch 'master' into xiao_esp32c6_with_ble

Ope8888 commented:
I can't wait for it to be finished :)

vidplace7 commented:
@alexbegoon If you rebase we'd be happy to review this 🙏

alexbegoon committed:
Merge remote-tracking branch 'upstream/master' into xiao_esp32c6_with_ble

 Conflicts:
	.gitignore
	bin/platformio-custom.py
	src/modules/AdminModule.cpp
	src/modules/Telemetry/EnvironmentTelemetry.cpp
	src/modules/Telemetry/PowerTelemetry.cpp
	src/nimble/NimbleBluetooth.cpp
	variants/esp32c6/tlora_c6/platformio.ini

alexbegoon committed:
Merge remote-tracking branch 'origin/xiao_esp32c6_with_ble' into xiao_esp32c6_with_ble

 Conflicts:
	arch/esp32/esp32c6.ini
	variants/tlora_c6/platformio.ini

alexbegoon commented:
I'm not sure if having ESP-IDF here is still relevant. I'm going to try building this using only the fork from Jason2866/platform-espressif32.
